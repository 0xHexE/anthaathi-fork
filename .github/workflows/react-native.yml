name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
