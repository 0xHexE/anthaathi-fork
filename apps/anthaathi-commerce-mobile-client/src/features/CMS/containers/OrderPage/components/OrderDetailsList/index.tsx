import {Pressable, View} from 'react-native';
import React from 'react';
import {Card, Text} from 'react-native-paper';
import {useIntl} from 'react-intl';
import {OrderPageComponentType} from '../../../../types/common';

export interface OrderDetailsProps {
  orderId: string;
  dateOfOrder: string;
  numberOrderItems: number;
  totalPrice: number;
  currency: string;
}

export interface OrderDetailsListProps {
  orders: OrderDetailsProps[];
}

const OrderDetailsList = (props: OrderDetailsListProps) => {
  return (
    <View style={{marginHorizontal: 10}} testID="orderDetailsList">
      {props.orders.map((order, index) => {
        return <OrderDetails key={index} order={order} />;
      })}
    </View>
  );
};

const OrderDetails = ({order}: {order: OrderDetailsProps}) => {
  const intl = useIntl();

  return (
    <Card
      style={{
        borderColor: '#E3E2E7',
        borderWidth: 1,
        borderRadius: 4,
        marginVertical: 5,
      }}
      testID="personalInformation">
      <Card.Content>
        <InformationRenderer
          label={intl.formatMessage({defaultMessage: 'Order ID'})}
          name={order.orderId}
        />
        <InformationRenderer
          label={intl.formatMessage({defaultMessage: 'Date'})}
          name={order.dateOfOrder}
        />
        <InformationRenderer
          label={intl.formatMessage({defaultMessage: 'Order Items'})}
          name={order.numberOrderItems + ' Items'}
        />
        <InformationRenderer
          label={intl.formatMessage({defaultMessage: 'Total Bill'})}
          name={intl.formatNumber(order.totalPrice, {
            style: 'currency',
            currency: order.currency,
          })}
        />
        <Pressable onPress={() => console.log('Pressed')}>
          <Text
            style={{
              color: '#008D3E',
              fontSize: 14,
              fontWeight: '700',
              marginVertical: 10,
              marginRight: 20,
            }}>
            {intl.formatMessage({defaultMessage: 'Reorder Items'})}
          </Text>
        </Pressable>
      </Card.Content>
    </Card>
  );
};

const InformationRenderer = ({label, name}: {label: string; name: string}) => {
  return (
    <View
      style={{
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginVertical: 5,
      }}>
      <Text
        testID="labelInfoRendererId"
        style={{color: '#808080', fontSize: 14, fontWeight: '400'}}>
        {label + ':  '}
      </Text>
      <Text
        testID="nameInfoRendererId"
        style={{color: '#364A15', fontSize: 14, fontWeight: '500'}}>
        {name}
      </Text>
    </View>
  );
};

export default OrderDetailsList;

export const OrderDetailsListCMSInput = {
  _component: OrderPageComponentType.OrderDetailsList,
  component: OrderDetailsList,
};
