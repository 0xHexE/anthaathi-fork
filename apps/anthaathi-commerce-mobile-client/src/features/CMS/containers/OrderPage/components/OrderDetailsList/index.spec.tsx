import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import OrderDetailsList from './index';

describe('OrderDetailsList', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <OrderDetailsList
            orders={[
              {
                orderId: '#123456',
                dateOfOrder: 'Tue, 12 Aug 2022',
                currency: 'USD',
                numberOrderItems: 5,
                totalPrice: 0,
              },
              {
                orderId: '#123456',
                dateOfOrder: 'Tue, 12 Aug 2022',
                currency: 'USD',
                numberOrderItems: 5,
                totalPrice: 0,
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('orderDetailsList')).toBeTruthy();
  });

  it('should have Order Details button title, labels and name', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <OrderDetailsList
            orders={[
              {
                orderId: '#123456',
                dateOfOrder: 'Tue, 12 Aug 2022',
                currency: 'USD',
                numberOrderItems: 5,
                totalPrice: 0,
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Reorder Items')).toBeTruthy();
  });

  it('should have number of information renderer', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <OrderDetailsList
            orders={[
              {
                orderId: '#123456',
                dateOfOrder: 'Tue, 12 Aug 2022',
                currency: 'USD',
                numberOrderItems: 5,
                totalPrice: 0,
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryAllByTestId('labelInfoRendererId').length).toBe(4);
    expect(temp.queryAllByTestId('nameInfoRendererId').length).toBe(4);
  });
});
