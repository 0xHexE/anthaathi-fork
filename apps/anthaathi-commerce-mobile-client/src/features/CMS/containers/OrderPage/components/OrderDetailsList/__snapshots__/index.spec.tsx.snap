// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetailsList should render the item 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 10,
    }
  }
  testID="orderDetailsList"
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 3,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "borderColor": "#E3E2E7",
            "borderRadius": 4,
            "borderWidth": 1,
            "elevation": 1,
            "marginVertical": 5,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexGrow": 1,
              "flexShrink": 1,
            }
          }
          testID="personalInformation"
        >
          <View
            style={
              Array [
                Object {
                  "paddingHorizontal": 16,
                },
                Object {
                  "paddingVertical": 16,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 5,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#808080",
                      "fontSize": 14,
                      "fontWeight": "400",
                    },
                  ]
                }
                testID="labelInfoRendererId"
              >
                Order ID:  
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#364A15",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                  ]
                }
                testID="nameInfoRendererId"
              >
                #123456
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 5,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#808080",
                      "fontSize": 14,
                      "fontWeight": "400",
                    },
                  ]
                }
                testID="labelInfoRendererId"
              >
                Date:  
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#364A15",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                  ]
                }
                testID="nameInfoRendererId"
              >
                Tue, 12 Aug 2022
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 5,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#808080",
                      "fontSize": 14,
                      "fontWeight": "400",
                    },
                  ]
                }
                testID="labelInfoRendererId"
              >
                Order Items:  
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#364A15",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                  ]
                }
                testID="nameInfoRendererId"
              >
                5 Items
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 5,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#808080",
                      "fontSize": 14,
                      "fontWeight": "400",
                    },
                  ]
                }
                testID="labelInfoRendererId"
              >
                Total Bill:  
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#364A15",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                  ]
                }
                testID="nameInfoRendererId"
              >
                $0.00
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#008D3E",
                      "fontSize": 14,
                      "fontWeight": "700",
                      "marginRight": 20,
                      "marginVertical": 10,
                    },
                  ]
                }
              >
                Reorder Items
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 3,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "borderColor": "#E3E2E7",
            "borderRadius": 4,
            "borderWidth": 1,
            "elevation": 1,
            "marginVertical": 5,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexGrow": 1,
              "flexShrink": 1,
            }
          }
          testID="personalInformation"
        >
          <View
            style={
              Array [
                Object {
                  "paddingHorizontal": 16,
                },
                Object {
                  "paddingVertical": 16,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 5,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#808080",
                      "fontSize": 14,
                      "fontWeight": "400",
                    },
                  ]
                }
                testID="labelInfoRendererId"
              >
                Order ID:  
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#364A15",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                  ]
                }
                testID="nameInfoRendererId"
              >
                #123456
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 5,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#808080",
                      "fontSize": 14,
                      "fontWeight": "400",
                    },
                  ]
                }
                testID="labelInfoRendererId"
              >
                Date:  
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#364A15",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                  ]
                }
                testID="nameInfoRendererId"
              >
                Tue, 12 Aug 2022
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 5,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#808080",
                      "fontSize": 14,
                      "fontWeight": "400",
                    },
                  ]
                }
                testID="labelInfoRendererId"
              >
                Order Items:  
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#364A15",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                  ]
                }
                testID="nameInfoRendererId"
              >
                5 Items
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 5,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#808080",
                      "fontSize": 14,
                      "fontWeight": "400",
                    },
                  ]
                }
                testID="labelInfoRendererId"
              >
                Total Bill:  
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#364A15",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                  ]
                }
                testID="nameInfoRendererId"
              >
                $0.00
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#008D3E",
                      "fontSize": 14,
                      "fontWeight": "700",
                      "marginRight": 20,
                      "marginVertical": 10,
                    },
                  ]
                }
              >
                Reorder Items
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
